src/geometry/PolySetBuilder.h

```diff
@@ -16,20 +16,26 @@ class PolySetBuilder
   void setConvexity(int n);
   int vertexIndex(const Vector3d& coord);
   int numVertices() const;
-  void appendPoly(int nvertices);
-  void append(const PolySet &ps);
-  void appendPoly(const std::vector<int>& inds);
+
+  void appendPolySet(const PolySet &ps);
   void appendGeometry(const std::shared_ptr<const Geometry>& geom);
-  void appendPoly(const std::vector<Vector3d>& v);
-  void appendVertex(int n);
-  void appendVertex(const Vector3d &v);
-  void prependVertex(int n);
-  void prependVertex(const Vector3d &v);
+  void appendPolygon(const std::vector<int>& inds);
+  void appendPolygon(const std::vector<Vector3d>& v);
+
+  void beginPolygon(int nvertices);
+  void addVertex(int n);
+  void addVertex(const Vector3d &v);
+  // Calling this is optional; will be called automatically when adding a new polygon or building the PolySet
+  void endPolygon();
```

Comment:
why should we have this if it can be called automatically?

Reply:
Really just for API cleanliness, as most client code will use begin/end pairs to make it more readable.

Reply:
We could make it required and rather assert if someone tries to do two beginPolygon() in a row.

Reply:
> We could make it required and rather assert if someone tries to do two beginPolygon() in a row.

Sounds good.